gammapy/data/observations.py

```diff
@@ -744,6 +744,19 @@ def from_stack(cls, observations_list):
         obs = itertools.chain(*observations_list)
         return cls(list(obs))
 
+    def iter(self, idx=None):
+        """Iterate over observations.
+
+        Parameters
+        ----------
+        idx : slice
+            Slice object defining which observations to return. Default is None, which means all observations.
+        """
+        if idx is None:
+            idx = slice(None)
+        for obs in self[idx]:
+            yield obs.copy(in_memory=True)
```

Comment:
do you need to explicitly delete the copy here? 
or do someting like 
```
obs_copy = obs.copy(in_memory=True)
yield obs_copy
```

Reply:
Don't know ...

Reply:
I think there is no need to trigger the `del` explicitly, as long as the observation goes out of scope. Which it typically does...